### ISSUE TITLE
Replace Youtube field with a new Embedded media field

### ISSUE THREAD

**Issue by sebastienros** [OrchardCMS/OrchardCore#14898]
Orchard 1 has one. 

https://learn.wordpress.org/tutorial/embedding-media-and-third-party-content-on-your-website/

**Comment by Piedone** [comment]
Why though? If I'm a non-technical user and want to embed a YouTube video, then I'll be more comfortable using something that very clearly is for YT. WP has this too, as also shown in the video.

**Comment by hishamco** [comment]
Because we need a field for each embedded media service for example Vimeo

**Comment by Piedone** [comment]
Then it's about adding an additional field, not replacing the YT one.

**Comment by Piedone** [comment]
Yes, and hence I'm saying https://github.com/OrchardCMS/OrchardCore/issues/14898#issuecomment-2123405841.

**Comment by sebastienros** [comment]
We decided to accept the YT field only temporarily at the beginning of the project. A more generic field is better suited and could embed from more that just YT. The effort from the user would be the same, provide a url, so there would be no point in having both.

**Comment by hishamco** [comment]
As I can see embedd can be anything Video, Audio, link, tweet .. etc, shall we support all types, by specifing the type in the admin, or showing the embedded link no matter what it was

**Comment by Piedone** [comment]
A more generic field is better in terms of DRY code and for us as developers because we like such swiss Army knife things, but I doubt a normal user would understand the Ultimate Video Embedder Wizard 3000 better than the "paste YouTube URL here" field :). And then again, WP also has a YT-specific field for some reason too.

But if anybody feels that this is important, why not, just make it an additional, not substitute, field.

**Comment by hishamco** [comment]
We could use a generic embedded field, and follow up to have something specific such as YouTube, Twitter .. etc

So, this means we should keep the YouTube field as its, but it SHOULD be inherited from a generic embedded field, if sound good I will start a PR

**Comment by Piedone** [comment]
Sounds good. I don't think this is high on the list of priorities though.

**Comment by hishamco** [comment]
Ya it's not high but at least we agree on that so we can plan then
